### ISSUE TITLE
False positive `no-member` in pylint 3.3.0 and higher

### ISSUE THREAD

**Issue by tsteijvers** [pylint-dev/pylint#10024]
### Bug description

The following code produces `E1101: Instance of 'str' has no 'result' member (no-member)` as of version 3.3.0 of pylint. In earlier versions of pylint the code was validated as correct. The `@task` decorator turns the function into a Prefect Task. When called with `return_state=True` the function returns a `State[str]`.

```
"""Test module"""
from prefect import flow, task
from prefect.states import Completed

@task
def task1(name: str) -> str:
    return f"Your name is {name}."

@flow
def myflow():
    result = task1("Jim", return_state=True)
    return Completed(message=result.result())
```

### Configuration
```
prefect==2.20.9
```

### Command used

`pylint myfile.py`

### Pylint output

E1101: Instance of 'str' has no 'result' member (no-member)

### Expected behavior

Code should be validated as correct since the return value of the called function is `State[str]`.

### Pylint version

pylint 3.3.0
astroid 3.3.5
Python 3.9.6 (default, Aug  9 2024, 14:24:13) 
[Clang 16.0.0 (clang-1600.0.26.3)]

### OS / Environment

MacOS Sonoma 14.7 (23H124)

**Comment by jacobtylerwalls** [comment]
Thanks for the report. While this may be a side effect of a recent change, the underlying missing feature is much broader and doc'd in #2578.
